commcare_connect/opportunity/tests/test_models.py

```diff
@@ -27,29 +27,37 @@ def test_learn_progress(opportunity: Opportunity):
 
 
 @pytest.mark.django_db
+@pytest.mark.parametrize("opportunity", [{}, {"opp_options": {"managed": True}}], indirect=True)
 def test_opportunity_stats(opportunity: Opportunity, user: User):
-    payment_unit_sub = PaymentUnitFactory(
+    payment_unit_sub = PaymentUnitFactory.create(
         opportunity=opportunity, max_total=100, max_daily=10, amount=5, parent_payment_unit=None
     )
-    payment_unit1 = PaymentUnitFactory(
+    payment_unit1 = PaymentUnitFactory.create(
         opportunity=opportunity,
         max_total=100,
         max_daily=10,
         amount=3,
         parent_payment_unit=payment_unit_sub,
     )
-    payment_unit2 = PaymentUnitFactory(
+    payment_unit2 = PaymentUnitFactory.create(
         opportunity=opportunity, max_total=100, max_daily=10, amount=5, parent_payment_unit=None
     )
     assert set(list(opportunity.paymentunit_set.values_list("id", flat=True))) == {
         payment_unit1.id,
         payment_unit2.id,
         payment_unit_sub.id,
     }
+    payment_units = [payment_unit_sub, payment_unit1, payment_unit2]
+    budget_per_user = sum(pu.max_total * pu.amount for pu in payment_units)
+    org_pay = 0
+    if opportunity.managed:
+        org_pay = opportunity.managedopportunity.org_pay_per_visit
+        budget_per_user += sum(pu.max_total * org_pay for pu in payment_units)
+    opportunity.total_budget = budget_per_user * 3
```

Comment:
This is to update the opportunity budget based on the Payment Units created earlier in the test and the `org_pay_per_visit` in case of a managed opportunity.

Reply:
It feels like there should be a nicer way to do this, but seems fine